ethereum/blockwatch/client.go

```diff
@@ -75,6 +76,10 @@ func (rc *RpcClient) HeaderByNumber(number *big.Int) (*meshdb.MiniHeader, error)
 	if err != nil {
 		return nil, err
 	}
+	// If it returned an empty struct, return NotFound error
+	if reflect.DeepEqual(header, reflect.Zero(reflect.TypeOf(header)).Interface()) {
```

Comment:
We don't need to use `reflect` to create the zero value if we already know the type ahead of time.

```suggestion
	if reflect.DeepEqual(header, GetBlockByNumberResponse{}) {
```

But even better, if there are some fields of `GetBlockByNumberResponse` that we know should always be present, we could just check those fields and avoid using reflect altogether. For example:

```go
if header.Number == "" {
    // ...
}
```

Reply:
Agreed. Let's go with `if header.Number == "" {` here.